src/promptflow-devkit/promptflow/_sdk/_tracing.py

```diff
@@ -428,17 +502,55 @@ def setup_exporter_to_pfs() -> None:
     _logger.debug("environ OTEL_EXPORTER_OTLP_ENDPOINT: %s", endpoint)
     if endpoint is not None:
         # create OTLP span exporter if endpoint is set
-        otlp_span_exporter = OTLPSpanExporter(endpoint=endpoint)
+        otlp_span_exporter = OTLPSpanExporterWithTraceURL(endpoint=endpoint)
         tracer_provider: TracerProvider = trace.get_tracer_provider()
         if not getattr(tracer_provider, TRACER_PROVIDER_PFS_EXPORTER_SET_ATTR, False):
             _logger.info("have not set exporter to prompt flow service, will set it...")
-            tracer_provider.add_span_processor(BatchSpanProcessor(otlp_span_exporter))
+            # Use a 1000 millis schedule delay to help export the traces in 1 second.
+            processor = BatchSpanProcessor(otlp_span_exporter, schedule_delay_millis=1000)
+            tracer_provider.add_span_processor(processor)
             setattr(tracer_provider, TRACER_PROVIDER_PFS_EXPORTER_SET_ATTR, True)
         else:
             _logger.info("exporter to prompt flow service is already set, no action needed.")
     _logger.debug("finish setup exporter to prompt flow service.")
 
 
+class OTLPSpanExporterWithTraceURL(OTLPSpanExporter):
+    PF_TRACE_URL_TEMPLATES = "PF_TRACE_URL_TEMPLATES"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._url_templates = self._load_url_templates()
+        self._printed_trace_ids = set()
+        self._lock = Lock()
+
+    def _load_url_templates(self):
+        try:
+            return json.loads(os.getenv(self.PF_TRACE_URL_TEMPLATES, "[]"))
+        except json.JSONDecodeError:
+            return []
+
+    def _print_trace_url(self, trace_id: str):
+        if not self._url_templates:
+            return
+        with self._lock:
+            #  Avoid printing the same trace URL multiple times
+            if trace_id in self._printed_trace_ids:
+                return
+            self._printed_trace_ids.add(trace_id)
+            print("You can view the trace detail from the following URL:")
```

Comment:
will this be too many log ? Any screen shot on final batch run or flow test output?

Reply:
Each trace_id would have one log, so flow test would be printed once, while batch run would be printed N times for N inputs.